### PR TITLE
Regression Tests Suite - AWS Code Build Infra - Initial Commit for #57

### PR DESCRIPTION
**Pull request by dhanainme**
* Script to clone Pytorch Repo & Run Post Man Scripts
* Basic Tests for Inference & Managment APIs
* AWS Code Build buildspec.yml

---

Attaching HTML Post Reports / Code Build Execution Logs 

```
start_torchserve /workspace//model_store /tmp/ts.log
--
953 | + torchserve --start --model-store /workspace//model_store --models /workspace//model_store/densenet161.mar
954 | + sleep 10
955 | + curl http://127.0.0.1:8081/models
956 | % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
957 | Dload  Upload   Total   Spent    Left  Speed
958 |  
959 | 0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
960 | 100   128  100   128    0     0   1969      0 --:--:-- --:--:-- --:--:--  1969
961 | {
962 | "models": [
963 | {
964 | "modelName": "densenet161",
965 | "modelUrl": "/workspace//model_store/densenet161.mar"
966 | }
967 | ]
968 | }
969 | + run_postman_test /tmp/test_exec.log
970 | + mkdir /workspace//report/
971 | + cd /codebuild/output/src670567202/src/github.com/pytorch/serve/test/
972 | + set +e
973 | + newman run -e postman/environment.json postman/mangement_api_test_collection.json -r html --reporter-html-export /workspace//report/api_report.html
974 | + newman run -e postman/environment.json postman/inference_api_test_collection.json -r html --reporter-html-export /workspace//report/management_report.html
975 | + set -e
976 | + cd -
977 | /workspace
978 | + stop_torch_serve
979 | + torchserve --stop
980 | TorchServe has stopped.
981 | + echo '** Tests Complete ** '
982 | ** Tests Complete **
983 | + exit 0
984 |  
985 | [Container] 2020/04/29 16:03:51 Phase complete: BUILD State: SUCCEEDED
986 | [Container] 2020/04/29 16:03:51 Phase context status code:  Message:
987 | [Container] 2020/04/29 16:03:51 Entering phase POST_BUILD
988 | [Container] 2020/04/29 16:03:51 Phase complete: POST_BUILD State: SUCCEEDED
989 | [Container] 2020/04/29 16:03:51 Phase context status code:  Message:
990 | [Container] 2020/04/29 16:03:51 Expanding base directory path: .
991 | [Container] 2020/04/29 16:03:51 Assembling file list
992 | [Container] 2020/04/29 16:03:51 Expanding .
993 | [Container] 2020/04/29 16:03:51 Expanding file paths for base directory .
994 | [Container] 2020/04/29 16:03:51 Assembling file list
995 | [Container] 2020/04/29 16:03:51 Expanding /workspace/report/api_report.html
996 | [Container] 2020/04/29 16:03:51 Expanding /workspace/report/management_report.html
997 | [Container] 2020/04/29 16:03:51 Expanding /tmp/ts.log
998 | [Container] 2020/04/29 16:03:51 Expanding /tmp/test_exec.log
999 | [Container] 2020/04/29 16:03:51 Found 4 file(s)
1000 | [Container] 2020/04/29 16:03:51 Updating artifact name as  TS-NIGHTLY-REGRESSION-2020-04-29-16:03
1001 | [Container] 2020/04/29 16:03:51 Phase complete: UPLOAD_ARTIFACTS State: SUCCEEDED
1002 | [Container] 2020/04/29 16:03:51 Phase context status code:  Message:
```

### CODE REVIEW

**Comment by lokeshgupta1975** [comment]
@dhanainme why is this not rebased on staging_0_1_1 branch?

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# TorchServe Regression Tests
+
+This folder contains nightly regression tests execututed against TorchServe master.These tests use [POSTMAN](https://www.postman.com/downloads/) for exercising all the Management & Inference APIs.
+
+### Latest Test Run Status
+
+![Build Status](https://codebuild.us-east-1.amazonaws.com/badges?uuid=eyJlbmNyeXB0ZWREYXRhIjoiS1QvY3lIUEdUb3hZVWNnbmJ2SEZCdExRNmNkNW9EVk1ZaFNldEk4Q0h3TU1qemwzQ29GNW0xMGFhZkxpOFpiMjUrZVVRVDUrSkh2ZDhBeFprdW5iNjRRPSIsIml2UGFyYW1ldGVyU3BlYyI6IjlvcjRqSTNMTmNhcExZbUwiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D&branch=master)
+
+[Test Logs](https://torchserve-regression-test.s3.amazonaws.com/torch-serve-regression-test/tmp/test_exec.log)
```
Comment (chauhang) [inline test/README.md:9]: @dhanainme Unable to access the s3 bucket -- getting access denied error so could not very the tests

Comment (dhanainme) [reply test/README.md:9]: CodeBuild seems to upload the log as private after every run. Just fixed the permissions. 
@chauhang , Do take a look & let me know if everything is fine.

**Comment by mycpuorg** [comment]
FWIW, this is the sort of output (just the inference snippet copied here.)
This is a promising start, thanks @dhanainme :

```
torchserve_regression_inference

→ Model Zoo - Register Text Classification
  POST http://localhost:8081/models?url=https://torchserve.s3.amazonaws.com/mar_files/my_text_classifier.mar&model_name=my_text_classifier&initial_workers=1&synchronous=true [200 OK, 299B, 8.6s]

  prepare   wait   dns-lookup   tcp-handshake   transfer-start   download   process   total 
  28ms      6ms    255µs        1ms             8.6s             5ms        510µs     8.6s  

--
→ Model Zoo - Inference Text Classification
  POST http://localhost:8080/predictions/my_text_classifier [200 OK, 239B, 114ms]

  prepare   wait    dns-lookup   tcp-handshake   transfer-start   download   process   total 
  4ms       841µs   66µs         544µs           105ms            6ms        64µs      118ms 

  ✓  Status code is 200

--
→ Model Zoo - Inference - DenseNet
  POST http://localhost:8080/predictions/densenet161 [200 OK, 467B, 557ms]

  prepare   wait    dns-lookup   tcp-handshake   transfer-start   download   process   total 
  2ms       411µs   (cache)      (cache)         554ms            1ms        79µs      558ms 

  ✓  Status code is 200

--
→ Model Zoo - Inference - SqueezeNet
  POST http://localhost:8080/predictions/squeezenet1_1 [200 OK, 459B, 288ms]

  prepare   wait    dns-lookup   tcp-handshake   transfer-start   download   process   total 
  1ms       331µs   (cache)      (cache)         285ms            1ms        53µs      289ms 

  ✓  Status code is 200

→ Inference Endpoint - Ping
  GET http://localhost:8080/ping [200 OK, 292B, 5ms]

  prepare   wait    dns-lookup   tcp-handshake   transfer-start   download   process   total 
  868µs     276µs   (cache)      (cache)         3ms              952µs      49µs      5ms   

  ✓  Status code is 200
```

File: test/README.md (hunk 1 above)
Comment (lokeshgupta1975) [reply test/README.md:9]: @chauhang Resolving this comment, please reopen it if you still have access issues.

**Comment by chauhang** [comment]
Thanks @dhaniram-kshirsagar This is a good start. Please go ahead and merge so that we can start testing the actual integration tests and build on top of this initial PR

File: test/README.md [hunk 2]
```diff
@@ -4,7 +4,7 @@ This folder contains nightly regression tests execututed against TorchServe mast
 
 ### Latest Test Run Status
 
-![Build Status](https://codebuild.us-east-1.amazonaws.com/badges?uuid=eyJlbmNyeXB0ZWREYXRhIjoiS1QvY3lIUEdUb3hZVWNnbmJ2SEZCdExRNmNkNW9EVk1ZaFNldEk4Q0h3TU1qemwzQ29GNW0xMGFhZkxpOFpiMjUrZVVRVDUrSkh2ZDhBeFprdW5iNjRRPSIsIml2UGFyYW1ldGVyU3BlYyI6IjlvcjRqSTNMTmNhcExZbUwiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D&branch=master)
+![Build Status](https://codebuild.us-east-1.amazonaws.com/badges?uuid=eyJlbmNyeXB0ZWREYXRhIjoiS1QvY3lIUEdUb3hZVWNnbmJ2SEZCdExRNmNkNW9EVk1ZaFNldEk4Q0h3TU1qemwzQ29GNW0xMGFhZkxpOFpiMjUrZVVRVDUrSkh2ZDhBeFprdW5iNjRRPSIsIml2UGFyYW1ldGVyU3BlYyI6IjlvcjRqSTNMTmNhcExZbUwiLCJtYXRlcmlhbFNldFNlcmlhbCI6MX0%3D&branch=staging_0_1_1)
```
Comment (mycpuorg) [inline test/README.md:7]: Does this mean this will only reflect staging branch's results for the tests?

Comment (mycpuorg) [reply test/README.md:7]: Perhaps, we should plan to add this badge for `master`

**Comment by dhanainme** [comment]
Nightly Test Runs have been running for a week : 

Logs : 

https://torchserve-regression-test.s3.amazonaws.com/public/latest/test_exec.log

Code Build : 

Screen Shot 2020-05-27 at 2 36 41 PM

**Pull request merged**